### PR TITLE
feat: add method to hash TX on `TransactionRequest` classes

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/1489
- Close: https://github.com/FuelLabs/fuels-ts/issues/1451

### CODE REVIEW

**Review by arboleya — DISMISSED** [review]

File: packages/providers/src/transaction-request/create-transaction-request.ts [hunk 1]
```diff
@@ -95,6 +96,10 @@ export class CreateTransactionRequest extends BaseTransactionRequest {
     );
   }
 
+  getTransactionId(chainId: number): string {
```
Comment (danielbate) [inline packages/providers/src/transaction-request/create-transaction-request.ts:99]: Doc block would be beneficial here as we gen API docs for this class.

File: apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md [hunk 2]
```diff
@@ -38,6 +38,14 @@ Note the method transfer has two parameters: the recipient's address and the int
 
 Once the predicate resolves with a return value `true` based on its predefined condition, our predicate successfully spends its funds by means of a transfer to a desired wallet.
 
+---
+
+In a similar approach, you can use the `createTransfer` method, which returns a `ScriptTransactionRequest`. Then, we can submit this transaction request by calling the `sendTransaction` method.
+
+This can be useful if you need the transaction ID before actually submitting it to the node.
```
Comment (danielbate) [inline apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md:45]: Wondering whether we need some docs here or _somewhere_ around how tx ids are formed, and the risk of it changing if the request is changed?

**Review by danielbate — DISMISSED** [review]
Left a couple of comments but will let you decide on priority

**Review by arboleya — COMMENTED** [review]
Should we consider merging this after `beta-5`?

Just wondering if some change could cascade into issues for the FE.

**Comment by Torres-ssf** [comment]
> Should we consider merging this after `beta-5`?
> 
> Just wondering if some change could cascade into issues for the FE.

We can definitely postpone this one. 

But the changes made here should not cause any breaking changes.

**Review by arboleya — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**